packages/openapi3/test/http-specs/specs.test.ts

```diff
@@ -0,0 +1,164 @@
+      "parameters/collection-format",
+
+      // warning @typespec/openapi3/xml-unwrapped-invalid-property-type: XML `@unwrapped` can only used on array properties or primitive ones in the OpenAPI 3 emitter, Property 'content' will be ignored.
+      // > 70 |   @unwrapped content: string;
+      "payload/xml",
+
+      // error @typespec/openapi3/unsupported-status-code-range: Status code range '494 to '499' is not supported. OpenAPI 3.0 can only represent range 1XX, 2XX, 3XX, 4XX and 5XX. Example: `@minValue(400) @maxValue(499)` for 4XX.
+      // 59 | model ErrorInRange {
+      "response/status-code-range",
+
+      // error @typespec/openapi3/path-query: OpenAPI does not allow paths containing a query string.
+      // > 435 |       op array(param: string[]): void;
+      "routes",
+    ],
+  });
+});
+
+describe("http-specs cases", () => {
+  worksFor(["3.0.0", "3.1.0"], ({ openApiForFile }) => {
+    describe("AuthApiKeyClient Rest Client", () => {
+      beforeAll(async () => {
+        exec("pnpm spector-serve", { cwd: pkgRoot });
+        await checkServe();
+      });
+
+      afterAll(() => {
+        execSync("pnpm spector-stop", { stdio: "inherit", cwd: pkgRoot });
+      });
+
+      it("should return 204 when the apiKey is valid", async () => {
+        const curr = {
+          name: "authentication/api-key",
+          fullPath: resolvePath(specsRoot, "authentication/api-key"),
+        };
+        const results = await openApiForFile(curr);
+        const res = Object.values(results)[0] as any;
+        const getThing = res.paths["/authentication/api-key/valid"].get;
+
+        const response = await markCoverage(`/authentication/api-key/valid`, {
+          method: "GET",
+          headers: {
+            "x-ms-api-key": "valid-key",
+          },
+        });
+        if (response) {
+          strictEqual(response?.status, 204);
+          ok(getThing.responses["204"]);
+        }
+      });
+
+      it("should return 403 when the apiKey is invalid", async () => {
+        const curr = {
+          name: "authentication/api-key",
+          fullPath: resolvePath(specsRoot, "authentication/api-key"),
+        };
+        const results = await openApiForFile(curr);
+        const res = Object.values(results)[0] as any;
+        const getThing = res.paths["/authentication/api-key/invalid"].get;
+        deepStrictEqual(getThing.responses["403"].content["application/json"].schema, {
+          $ref: "#/components/schemas/InvalidAuth",
+        });
+        const invalidAuthSchemas = res.components.schemas;
+
+        const response = await markCoverage(`/authentication/api-key/invalid`, {
+          method: "GET",
+          headers: {
+            "x-ms-api-key": "invalid-key",
+          },
+        });
+        if (response) {
+          strictEqual(response.status, 403);
+          strictEqual(response.body.error, "invalid-api-key");
+          await validataDataWithSchema(response.body, invalidAuthSchemas);
```

Comment:
Looks like we should remove unrelated schema for 200.
Suppose `components/schemas` contains different schemas: A, B. Reponses is A, but we expect B, then we still passes the validation.